### ISSUE TITLE
Don't upgrade while background jobs are currently being executed

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/server#24777]
The updater needs to check that no background job is reserved at the moment. (maybe it could even reserve them all?)
Happened on c.nc.c and purged all rooms because attendees where currently being migrationed but all rooms with out attendees where deleted in parallel with the background job that was started ~6 minutes before and didn't finish nor stop to take new tasks while the upgrade was running in parallel.

**Comment by ggeorgg** [comment]
When I use the command line to put nextcloud into maintenance mode with `sudo -u nginx php7 /srv/www/nextcloud/occ maintenance:mode --on` I (almost) always get the following error message logged:
![grafik](https://user-images.githubusercontent.com/13244552/110230411-0e899000-7f11-11eb-85a2-f424318b0b02.png)

The same is true, when I do update apps via the Web GUI. 

I guess there are currently running requests which can't be finished when the maintenance mode is enabled. How could I avoid this error message in the log?

Not sure if the described problem is related to the issue... Should I open a new issue?

**Comment by DanScharon** [comment]
This seems rather important and I thought that would have been implemented ages ago. Quite unfortunate that this is getting pushed back milestone after milestone.

**Comment by joshtrichards** [comment]
Related:

- #45513 
- #28508
